tt_eager/tt_dnn/op_library/composite/composite_ops.cpp

```diff
@@ -1529,13 +1529,27 @@ Tensor pow(const Tensor& input_a, int exponent, const MemoryConfig& output_mem_c
     return power(input_a, exponent, output_mem_config);
 }
 
+Tensor create_mask(const Tensor& input_a, const MemoryConfig& output_mem_config)
+{
+    auto& padded_shape = input_a.get_legacy_shape();
+    auto& unpadded_shape = padded_shape.without_padding();
+
+    if (padded_shape == unpadded_shape)
+        return input_a;
+    float t_inf = -std::numeric_limits<float>::infinity();
+    Tensor masked_input = tt::numpy::mask_padded_input<bfloat16>(padded_shape, unpadded_shape, DataType::BFLOAT16);
+    masked_input = where(eqz(masked_input, output_mem_config), t_inf, input_a, output_mem_config);
```

Comment:
If we are creating the masked tensor ourselves, there should be no need to run this where and eqz op on device, just directly create the tensor with the correct mask value

Reply:
@tt-aho when the user pad the tensor with pad value (it might affect argmax functionality as it checks for max value and returns index), in order to ignore the user's  pad value , we created an mask_tensor of 1 and 0 and replace 0 by -inf and 1 by the actual input value.

user input (pad value here is 100)

[ 5.,  3.,  2.,  1., 10.,  9.,  6.,  7.,  4.,  8.,  100. , 100., 100., 100. 100. ........]

after mask with where and eqz check, input for argmax will be

[ 5.,  3.,  2.,  1., 10.,  9.,  6.,  7.,  4.,  8.,  -inf , -inf, -inf, -inf -inf........]

I doubt whether this will be possible while creating mask tensor itself

Reply:
I see. I'm not sure why you need to run `eqz` then, isn't this just flipping the 1s and 0s you originally make in the mask tensor? So if you create the mask with the inverted values then there is no need for the eqz?

Reply:
Updated and removed the eqz op usage.